### PR TITLE
Add panels

### PR DESCRIPTION
**Pull request by ptbrowne**
### Goal

- Useful to display a main content along a side content.

### API 
- `Panel.Group`
- `Panel.Main`
- `Panel.Side`

It is showcased in the modal here : https://ptbrowne.github.io/cozy-ui/react/

### CODE REVIEW

File: stylus/components/panels.styl [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+$panel-group
+    display         flex
+    flex-direction  row
+    justify-content flex-start
+    align-items     stretch
+    box-sizing      border-box
+
+$panel-main
+    flex-basis 65%
```
Comment (GoOz) [inline stylus/components/panels.styl]: You could use the shorhand here. `flex 1 0 65%`

File: stylus/components/panels.styl [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    flex-grow 1
+    flex-shrink 0
+    overflow auto
+
+$panel-side
+    flex-basis 35%
```
Comment (GoOz) [inline stylus/components/panels.styl]: If I remember correctly some browsers don't have the same default values for `shrink` and/or `grow` and you might as well write them explicitely with `flex 0 0 35%`

File: react/Panel/styles.styl [hunk 3]
```diff
@@ -0,0 +1,10 @@
+@require '../../stylus/components/panels'
+
+.Panel-group
```
Comment (GoOz) [inline react/Panel/styles.styl]: Is the nesting really necessary? It doesn't do much besides inscreasing the specificity I think.

Comment (ptbrowne) [reply react/Panel/styles.styl]: You're right.

**Pull request merged**